app/redux/wallet/actions.ts

```diff
@@ -109,16 +110,26 @@ export const createNewWallet = ({
   genesisID,
   type,
   name,
+  mnemonicType,
 }: {
   existingMnemonic?: string | undefined;
   password: string;
   type: WalletType;
   apiUrl: SocketAddress | null;
   genesisID: string;
   name?: string;
+  mnemonicType: MnemonicStrengthType;
```

Comment:
Now we have two mutually exclusive properties: `existingMnemonic` and `mnemonicType`. But it is not clear from the interface, and `mnemonicType` is redundantly pushing the User to call `existingMnemonic.length` there, while this variable won't be used at all.

I'd prefer to have a clear interface.
Something like this:

```suggestion
  mnemonics: { generate: MnemonicStrengthType } | { existing: string };
  password: string;
  type: WalletType;
  apiUrl: SocketAddress | null;
  genesisID: string;
  name?: string;
```